cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -178,13 +179,26 @@ public void getArtifacts(HttpRequest request, HttpResponder responder,
   public void getArtifactVersions(HttpRequest request, HttpResponder responder,
                                   @PathParam("namespace-id") String namespaceId,
                                   @PathParam("artifact-name") String artifactName,
-                                  @QueryParam("scope") @DefaultValue("user") String scope)
+                                  @QueryParam("scope") @DefaultValue("user") String scope,
+                                  @QueryParam("versionRange") String versionRange,
+                                  @QueryParam("limit") @DefaultValue("-1") String limit,
+                                  @QueryParam("order") String order)
     throws Exception {
 
     NamespaceId namespace = validateAndGetScopedNamespace(Ids.namespace(namespaceId), scope);
 
+    ArtifactRange range = versionRange == null ? null : ArtifactRange.parse(namespace, artifactName, versionRange);
+    int limitNumber = Integer.valueOf(limit);
+    SortInfo.SortOrder sortOrder = order == null ? null : SortInfo.SortOrder.valueOf(order);
```

Comment:
in any case, this is out of the scope of this PR, as SortInfo looks like its used by a bunch of metadata classes. I think we should be using a different enum though, to avoid confusion.